clickControl/clickControl.py

```diff
@@ -1,73 +1,124 @@
+        # woe be those that understand the MAGIMessage or why the
+        # args is a single value in a tuple...
+        func_inputs = args[0].__dict__['data'].split('\n')[0][5:].strip()
+        # this is a hack
+        replacer = re.compile("([a-zA-Z0-9_]+)")
+        func_inputs = replacer.sub(r'"\1"', func_inputs)
+        func_args = ast.literal_eval(
+            func_inputs
+        )
+        # in order to get the configuration, we will need to make
+        # sure we have parsed conf file (not guarenteed at time of
+        # calling this function).  Also this will all break if the
+        # confinguration file is not in '/click', but that is not
+        # something that is allowed to be changed in clickControl
+        self.ccp.parse()
+        config = self.ccp.get_configuration()
+        # verify node is valid
+        valid_node = config.get(func_args['node'], False)
+        if not valid_node:
+            raise ClickControlError(
+                'NODE: {user_node} for click object not found. ' \
+                'valid key targets are: {config_node_keys}\n'.format(
+                    user_node=func_args['node'],
+                    config_node_keys=config.keys(),
+                )
+            )
+        valid_key = valid_node.get(func_args['key'], False)
+        if not valid_key:
+            raise ClickControlError(
+                'KEY: {user_key} for click object: {node} was not found. '\
+                'valid key targets are: {config_user_keys}\n'.format(
+                    user_key=func_args['key'],
+                    node=func_args['node'],
+                    config_user_keys=valid_node.keys(),
+                )
+            )
+        return orig_func(self, *args, **kwargs)
+    return wrapper
+
+
+# pylint: disable=too-many-instance-attributes
+class ClickControlAgent(DispatchAgent):
     """
     This click control agent allows the dynamic control of click-based network emulation
     """
     def __init__(self):
         DispatchAgent.__init__(self)
         self.log = logging.getLogger(__name__)
         self.click_config = "/tmp/vrouter.click"
-        self.UDPRunning = False
-        self.isFlapping = False
+        self.udp_running = False
+        self.is_flapping = False
         self.ccp = ClickConfigParser()
-
-
-        self.cg = clickGraph.clickGraph(self.click_config)
+        self.clg = clickGraph.clickGraph(self.click_config)
         # assumes clicks installed, should we install?
-
-        self._clickProc = None
-        self._confPath = '/click'   # handle to click's runtime configuration.
+        self._click_proc = None
+        self._conf_path = '/click'   # handle to click's runtime configuration.
 
     @agentmethod()
+    # pylint: disable=unused-argument,no-self-use
     def updateVisualization(self, msg):
         return True
 
     @agentmethod()
-    def startClick(self, msg, userMode=False, DPDK=True):
+    def startClick(self, msg, user_mode=False, dpdp=True):
```

Comment:
dpdk=True may be better

Reply:
not sure what you mean by this comment.  It is already set to True?  Is this a snake case vs camel case thing?

Reply:
I mean "dpdk" vs. "dpdp". DPDK is "Data Plane Development Kit" and stands for that.

Reply:
thats my stupidity acting up =)
I had just corrected erik on SRIOV yesterday, but dkdp is too many d's too close for me to handle.  I'll fix it up.